src/common/Props.js

```diff
@@ -97,21 +121,66 @@ export default class Props extends Map {
 		return prop;
 	}
 
+	/**
+	 * Add a prop regardless of whether props have been processed or not yet
+	 * @param {Function} Class
+	 * @param {string} name
+	 * @param {object} spec
+	 */
+	static add (Class, name, spec) {
+		if (Class.props instanceof this) {
+			Class.props.add(name, spec);
+		}
+		else {
+			Class.props[name] = spec;
+		}
+	}
+
 	updateDependents () {
 		if (!this.#initialized) {
 			// We update all dependents at once after initialization
+			// no need to do it after every single property
 			return;
 		}
 
-		// Build or update dependency graph
+		// Rebuild dependency graph
+		let dependents = {};
+
+		for (let name of this.keys()) {
+			dependents[name] = new Set();
+		}
+
+		let keyIndices = Object.fromEntries(this.keys().map((key, i) => [key, i]));
```

Comment:
For some reason, this line throws an error in Firefox: `Uncaught TypeError: this.keys().map is not a function`.

Safari fails silently. 🤬

Reply:
```suggestion
		let keyIndices = Object.fromEntries([...this.keys()].map((key, i) => [key, i]));
```

Reply:
It does! Yay! 🥳

Suggested the change.

Reply:
Yeah, that's because adding `map` and friends to iterables is pretty new